### ISSUE TITLE
Temporal Accumulation Errors from Temporal Autoregressive Model 

### ISSUE THREAD

**Issue by MinghanLi** [jy0205/Pyramid-Flow#214]
The temporal autoregressive approach introduces temporal accumulation errors, leading to issues such as blurriness and darker tones in videos, especially for longer sequences. Additional artifacts may also appear over time. Below are two examples of 16-frame video generation at 384p resolution: the first video shows results from MMDIT, while the second video shows results from MiniFlux. 

https://github.com/user-attachments/assets/6335994e-d5e3-4dda-84d4-4e35a74008e2

https://github.com/user-attachments/assets/660b1861-643b-4600-bdcb-cf8fb5d57768

A possible solution to address this issue is to truncate the historical frame information, such as using only the previous 8 frames for autoregression. Have there been any plans or considerations for implementing this solution in future work?

**Comment by zhichengsun** [comment]
Yes, temporal accumulation error is an important problem we will look into in the future work. We recommend reading [CausVid](https://arxiv.org/abs/2412.07772) as a highly relevant work.

**Comment by feifeiobama** [comment]
Yes, temporal error accumulation is an important problem we will look into in the future work. We recommend reading [CausVid](https://arxiv.org/abs/2412.07772) as a highly relevant work.

**Comment by zlenyk** [comment]
@feifeiobama Could I follow-up on this thread? I was recently experimenting with your code in the direction that @MinghanLi  mentioned - keeping only K last frames to avoid conditioning on extremely long tensors. However I found this to collapse the outputs rather quickly (a bit counter-intuitive for me). Do you have any intuition why this might be happening?

I don't think there should be any issue with positional embeddings, as we are just pretending that the earliest latent is start of the video.

I'm providing 2 videos (both generating temp=20 latents). One keeps all the historical frames, 2nd keeps only last 4 latents (which should be equivalent to 4x8=32 frames).

https://github.com/user-attachments/assets/cf727e91-1e77-47b1-a665-ea43c44036c6
https://github.com/user-attachments/assets/ec9d4a16-54ba-4295-8073-21beb3b94cbe

**Comment by feifeiobama** [comment]
I think this could be related to Causal VAE where the first latent frame is different from the subsequent ones. To apply sliding window, one might have to finetune the model a little bit.
